tensorflow_datasets/image/arc.py

```diff
@@ -0,0 +1,126 @@
+# coding=utf-8
+# Copyright 2019 The TensorFlow Datasets Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""ARC dataset."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from absl import logging
+import json
+import numpy as np
+import os
+import tensorflow as tf
+import tensorflow_datasets.public_api as tfds
+
+_CITATION = """
+@misc{chollet_françois_2019,
+  title     = {The Measure of Intelligence},
+  url       = {https://arxiv.org/abs/1911.01547},
+  journal   = {arXiv.org},
+  author    = {François Chollet},
+  year      = {2019},
+  month     = {Nov}
+}
+"""
+
+_DESCRIPTION = """
+ARC can be seen as a general artificial intelligence benchmark, as a program
+synthesis benchmark, or as a psychometric intelligence test. It is targeted at
+both humans and artificially intelligent systems that aim at emulating a
+human-like form of general fluid intelligence.
+"""
+
+
+_BASE_URL = "https://github.com/fchollet/ARC/"
+_COMMIT = "bd9e2c934c83d00251b7b4781ffc38cd167c885f"
```

Comment:
As the data seems to be evolving, the user should be able to select the version he wants.
Please use `BuilderConfig` which contains the commit, so that users is not forced to use the default commit.

Reply:
I just made this change, works fine. However, I'm not sure how to handle the checksums, since each commit will have a different checksum. Will the user have to run the `tensorflow_datasets.scripts.download_and_prepare` script with the `--register_checksums` option the first time they use a new commit? How exactly would that work? Would they have to define a new dataset?